src/main/java/org/jabref/gui/FrameDndHandler.java

```diff
@@ -0,0 +1,198 @@
+package org.jabref.gui;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.function.Supplier;
+
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.Tab;
+import javafx.scene.control.TabPane;
+import javafx.scene.control.skin.TabPaneSkin;
+import javafx.scene.input.DragEvent;
+import javafx.scene.input.Dragboard;
+import javafx.scene.input.TransferMode;
+
+import org.jabref.gui.importer.actions.OpenDatabaseAction;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.groups.GroupTreeNode;
+
+import com.tobiasdiez.easybind.EasyBind;
+
+public class FrameDndHandler {
+    private final TabPane tabPane;
+    private final Supplier<Scene> scene;
+    private final Supplier<OpenDatabaseAction> openDatabaseAction;
+    private final StateManager stateManager;
+
+    public FrameDndHandler(TabPane tabPane,
+                           Supplier<Scene> scene,
+                           Supplier<OpenDatabaseAction> openDatabaseAction,
+                           StateManager stateManager) {
+        this.tabPane = tabPane;
+        this.scene = scene;
+        this.openDatabaseAction = openDatabaseAction;
+        this.stateManager = stateManager;
+    }
+
+    void initDragAndDrop() {
+        tabPane.setTabDragPolicy(TabPane.TabDragPolicy.REORDER);
+
+        Tab dndIndicator = new Tab(Localization.lang("Open files..."), null);
```

Comment:
I think, there should be a constructor being without the need to pass `null`. That case should be handled by the internal workings and not by the external caller.

Reply:
Indeed, for a follow up pr about streamlining opening databases even more, but would be out of scope of this pr. Needs more thought and larger refactorings.